src/CTA.Rules.Models/Tokens/ExpressionToken.cs

```diff
@@ -0,0 +1,17 @@
+﻿using CTA.Rules.Config;
+
+namespace CTA.Rules.Models.Tokens
+{
+    public class ExpressionToken : NodeToken
+    {
+        public override bool Equals(object obj)
+        {
+            var token = (ExpressionToken)obj;
+            return token.Key == this.Key && token.Namespace == this.Namespace && token.Type == this.Type;
```

Comment:
Any null checks needed here?

Reply:
Added checks to be safe on the safe side across the other objects too.